### PR TITLE
fix: const extGw may expired, after subnet updated, so use ipam subne…

### PR DESCRIPTION
**Pull request by bobz965**
fix:  After external subnet updated, const extGw may expired, so use ipam subnet to cache ipv4|v6 gw。

also,  later ipv6(dualstack) eip and route in nat-gateway.sh  need another set of ipv6 route,eip add function, ipam subnet cache gw would be more convenient

- [x] Make sure you have followed [Kube-OVN Code Style](../CODE_STYLE.md).

#### What type of this PR
- Bug fixes

### CODE REVIEW

File: pkg/controller/vpc_nat_gateway.go [hunk 1]
```diff
@@ -533,26 +534,21 @@ func (c *Controller) handleUpdateNatGwSubnetRoute(natGwKey string) error {
...
+	extRules = append(extRules, fmt.Sprintf("%s,%s", v4ExternalCidr, v4ExternalGw))
+	if err = c.execNatGwRules(pod, natGwExtSubnetRouteAdd, extRules); err != nil {
+		klog.Errorf("failed to exec nat gateway rule, err: %v", err)
 		return err
 	}
 
-	gwSubnet, err := c.subnetsLister.Get(gw.Spec.Subnet)
-	if err != nil {
-		klog.Errorf("failed to get subnet, err: %v", err)
+	if v4InternalGw, _, err = c.GetGwbySubnet(gw.Spec.Subnet); err != nil {
```
Comment (oilbeater) [inline pkg/controller/vpc_nat_gateway.go:551]: should log error here to follow the code style

File: pkg/controller/vpc_nat_gw_eip.go [hunk 2]
```diff
@@ -249,11 +248,11 @@ func (c *Controller) handleAddIptablesEip(key string) error {
 	if eipV4Cidr, err = c.getEipV4Cidr(v4ip); err != nil {
 		return err
 	}
-	if gw, err = c.GetGw(); err != nil {
+	if v4Gw, _, err = c.GetGwbySubnet(util.VpcExternalNet); err != nil {
```
Comment (oilbeater) [inline pkg/controller/vpc_nat_gw_eip.go:251]: Also log error here to fix the legacy style issues

File: pkg/controller/vpc_nat_gw_eip.go [hunk 3]
```diff
@@ -496,12 +495,12 @@ func (c *Controller) handleUpdateIptablesEip(key string) error {
...
 			return err
 		}
-		gw, err := c.GetGw()
+		v4Gw, _, err := c.GetGwbySubnet(util.VpcExternalNet)
 		if err != nil {
 			klog.Errorf("failed to get gw, %v", err)
 			return err
 		}
-		err = c.createEipInPod(eip.Spec.NatGwDp, gw, eipV4Cidr)
+		err = c.createEipInPod(eip.Spec.NatGwDp, v4Gw, eipV4Cidr)
```
Comment (oilbeater) [inline pkg/controller/vpc_nat_gw_eip.go:503]: And here

File: pkg/controller/vpc_nat_gw_eip.go [hunk 4]
```diff
@@ -249,11 +248,11 @@ func (c *Controller) handleAddIptablesEip(key string) error {
 	if eipV4Cidr, err = c.getEipV4Cidr(v4ip); err != nil {
 		return err
 	}
-	if gw, err = c.GetGw(); err != nil {
+	if v4Gw, _, err = c.GetGwbySubnet(util.VpcExternalNet); err != nil {
 		return err
 	}
 	// create
-	err = c.createEipInPod(eip.Spec.NatGwDp, gw, eipV4Cidr)
+	err = c.createEipInPod(eip.Spec.NatGwDp, v4Gw, eipV4Cidr)
```
Comment (oilbeater) [inline pkg/controller/vpc_nat_gw_eip.go:255]: Legacy style issues here

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/vpc_nat_gateway.go (hunk 1 above)
Comment (bobz965) [reply pkg/controller/vpc_nat_gateway.go:551]: done

File: pkg/controller/vpc_nat_gw_eip.go (hunk 2 above)
Comment (bobz965) [reply pkg/controller/vpc_nat_gw_eip.go:251]: done

File: pkg/controller/vpc_nat_gw_eip.go (hunk 3 above)
Comment (bobz965) [reply pkg/controller/vpc_nat_gw_eip.go:500]: done

File: pkg/controller/vpc_nat_gw_eip.go (hunk 4 above)
Comment (bobz965) [reply pkg/controller/vpc_nat_gw_eip.go]: done

**Pull request merged**